docs/TypeChecker.rst

```diff
@@ -971,3 +971,34 @@ well-typed suggestions.
   meets and joins, so the solver continues until it runs out of
   supertypes to enumerate.
 
+New Diagnostic Architecture
+---------------------------
+
+We are currently working on porting type-check based diagnostics over
+to the new diagnostic framework, which is described in detail in this
+`blog post
+<https://swift.org/blog/new-diagnostic-arch-overview/>`_.
+
+The things in the queue yet to be ported are:
+
+- ``visitTupleExpr``: Diagnostics related to label/type mismatches
```

Comment:
Should we remove this since I'm working on it now?

Reply:
I think it would make sense to remove once the work is complete?